addon-test-support/@ember/test-helpers/dom/fill-in.ts

```diff
@@ -72,6 +72,13 @@ export default function fillIn(target: Target, text: string): Promise<void> {
           '`fillIn` is only usable on form controls or contenteditable elements.'
         );
       }
+
+      return settled();
+    })
+    .then(() => runHooks('fillIn', 'before-fire-event', target, text))
```

Comment:
I was thinking that you would actually add this to `fireEvent` itself, that way you only haev to change one file (not all of hte dom helpers). 

Is there any reason that wouldn't work?

Reply:
So I'm now deep into this thing, and thought I should check in about my progress and confirm that the direction I'm heading is kosher. 

Specifically, given the hooks system uses Promises, I've been forced to "Promisify" any function where `fireEvent` is called. Given the library forbids the use of `async`/`await`, in places, the changes are fairly substantial and it's not easy to keep things readable. It'd doable, though.

My assumption about `async`/`await` is that we forbid it for backwards compatibility with older apps. Is that correct?

Anyway, I'm about 60–80% of the way done now.